The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/22867317030/job/66337595583) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/66337595583) [(plain)](https://github.com/rust-lang/rust/commit/4f2f3f32be0950548a312a5e3b777b9edc185e43/checks/66337595583/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
/dev/sda15       98M  6.4M   92M   7% /boot/efi
tmpfs           1.6G   16K  1.6G   1% /run/user/1001
================================================================================

Sufficient disk space available (120898984KB >= 52428800KB). Skipping cleanup.
##[group]Run echo "[CI_PR_NUMBER=$num]"
echo "[CI_PR_NUMBER=$num]"
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
set -ex

# Run a subset of tests. Used to run tests in parallel in multiple jobs.

# When this job partition is run as part of PR CI, skip tidy to allow revealing more failures. The
# dedicated `tidy` job failing won't block other PR CI jobs from completing, and so tidy failures
# shouldn't inhibit revealing other failures in PR CI jobs.
if [ "$PR_CI_JOB" == "1" ]; then
  echo "PR_CI_JOB set; skipping tidy"
  SKIP_TIDY="--skip tidy"
fi

../x.py --stage 2 test \
  ${SKIP_TIDY:+$SKIP_TIDY} \
  --skip compiler \
  --skip src
#!/bin/bash

set -ex

# Run a subset of tests. Used to run tests in parallel in multiple jobs.

# When this job partition is run as part of PR CI, skip tidy to allow revealing more failures. The
# dedicated `tidy` job failing won't block other PR CI jobs from completing, and so tidy failures
# shouldn't inhibit revealing other failures in PR CI jobs.
if [ "$PR_CI_JOB" == "1" ]; then
  echo "PR_CI_JOB set; skipping tidy"
  SKIP_TIDY="--skip tidy"
fi

../x.py --stage 2 test \
  ${SKIP_TIDY:+$SKIP_TIDY} \
  --skip tests \
  --skip coverage-map \
  --skip coverage-run \
  --skip library \
  --skip tidyselftest
---
##[endgroup]
Executing "/scripts/stage_2_test_set1.sh"
+ /scripts/stage_2_test_set1.sh
+ '[' 1 == 1 ']'
+ echo 'PR_CI_JOB set; skipping tidy'
+ SKIP_TIDY='--skip tidy'
+ ../x.py --stage 2 test --skip tidy --skip compiler --skip src
PR_CI_JOB set; skipping tidy
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
---
test [ui] tests/ui/abi/abi-sysv64-register-usage.rs ... ignored, ignored when the architecture is aarch64
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_disabled ... ok
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_enabled ... ok
test [ui] tests/ui/abi/anon-extern-mod.rs ... ok
test [ui] tests/ui/abi/avr-sram.rs#disable_sram ... ok
test [ui] tests/ui/abi/avr-sram.rs#has_sram ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#aarch64 ... ok
test [ui] tests/ui/abi/avr-sram.rs#no_sram ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#arm ... ok
test [ui] tests/ui/abi/bad-custom.rs ... ok
test [ui] tests/ui/abi/c-stack-as-value.rs ... ok
test [ui] tests/ui/abi/c-zst.rs#aarch64-darwin ... ok
test [ui] tests/ui/abi/c-zst.rs#powerpc-linux ... ok
---
test [ui] tests/ui/asm/aarch64/ttbr0_el2.rs ... ok
test [ui] tests/ui/asm/aarch64/type-check-2.rs ... ok
test [ui] tests/ui/asm/aarch64/type-check-3.rs ... ok
test [ui] tests/ui/asm/aarch64/type-f16.rs ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#hf ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#r82 ... ok
test [ui] tests/ui/asm/aarch64/sym.rs ... ok
test [ui] tests/ui/asm/arm-low-dreg.rs ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#sf ... ok
test [ui] tests/ui/asm/binary_asm_labels.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/asm/asm-with-nested-closure.rs ... ok
test [ui] tests/ui/asm/bad-template.rs#aarch64 ... ok
test [ui] tests/ui/asm/cfg.rs#reva ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/asm/cfg.rs#revb ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/const-generics/occurs-check/unused-substs-2.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-3.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-4.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-5.rs ... ok
test [ui] tests/ui/const-generics/ogca/basic.rs ... ok
test [ui] tests/ui/const-generics/ogca/basic-fail.rs ... ok
test [ui] tests/ui/const-generics/ogca/generic-param-rhs.rs ... ok
test [ui] tests/ui/const-generics/ogca/coherence-ambiguous.rs ... ok
test [ui] tests/ui/const-generics/ogca/rhs-but-not-root.rs ... ok
test [ui] tests/ui/const-generics/outer-lifetime-in-const-generic-default.rs ... ok
test [ui] tests/ui/const-generics/opaque_types.rs ... ok
test [ui] tests/ui/const-generics/overlapping_impls.rs ... ok
test [ui] tests/ui/const-generics/opaque_types2.rs ... ok
test [ui] tests/ui/const-generics/params-in-ct-in-ty-param-lazy-norm.rs#full ... ok
---
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#fat2 ... ok
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#fat3 ... ok
test [ui] tests/ui/extern/issue-80074.rs ... ok
test [ui] tests/ui/extern/issue-95829.rs ... ok
test [ui] tests/ui/extern/lgamma-linkage.rs ... ok
test [ui] tests/ui/extern/no-mangle-associated-fn.rs ... ok
test [ui] tests/ui/extern/not-in-block.rs ... ok
test [ui] tests/ui/extern/unsized-extern-derefmove.rs ... ok
test [ui] tests/ui/extern/windows-tcb-trash-13259.rs ... ok
test [ui] tests/ui/feature-gates/allow-features-empty.rs ... ok
---
test [ui] tests/ui/feature-gates/feature-gate-macro-derive.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-macro-metavar-expr-concat.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-marker_trait_attr.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-may-dangle.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-mgca-type-const-syntax.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-min-generic-const-args.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-movrs_target_feature.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/feature-gates/feature-gate-more-maybe-bounds.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-min_const_fn.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-naked_functions_rustic_abi.rs ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/imports/ambiguous-9.rs ... ok
test [ui] tests/ui/imports/ambiguous-import-visibility-module.rs ... ok
test [ui] tests/ui/imports/ambiguous-8.rs ... ok
test [ui] tests/ui/imports/ambiguous-glob-vs-expanded-extern.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-glob-vs-multiouter.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-globvsglob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-no-implicit-prelude.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-core-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-std-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-pick-core.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-pick-std.rs ... ok
---
test [ui] tests/ui/layout/post-mono-layout-cycle.rs ... ok
test [ui] tests/ui/layout/randomize.rs#normal ... ok
test [ui] tests/ui/layout/randomize.rs#randomize-layout ... ok
test [ui] tests/ui/layout/null-pointer-optimization.rs ... ok
test [ui] tests/ui/layout/rigid-alias-due-to-broken-impl.rs ... ok
test [ui] tests/ui/layout/size-of-val-raw-too-big.rs ... ignored, only executed when the pointer width is 32bit (Layout computation rejects this layout for different reasons on 64-bit.)
test [ui] tests/ui/layout/reprc-power-alignment.rs ... ok
test [ui] tests/ui/layout/struct.rs ... ok
test [ui] tests/ui/layout/rigid-alias-no-params.rs ... ok
test [ui] tests/ui/layout/rust-call-abi-not-a-tuple-ice-81974.rs ... ok
---
To only update this specific test, also pass `--test-args feature-gates/issue-43106-gating-of-test.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/feature-gates/issue-43106-gating-of-test.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/issue-43106-gating-of-test" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
warning: lint `soft_unstable` has been removed: the general soft-unstable mechanism has been removed
##[warning]  --> /checkout/tests/ui/feature-gates/issue-43106-gating-of-test.rs:3:10
   |
---
To only update this specific test, also pass `--test-args imports/issue-28134.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/imports/issue-28134.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/imports/issue-28134" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--test"
stdout: none
--- stderr -------------------------------
warning: lint `soft_unstable` has been removed: the general soft-unstable mechanism has been removed
##[warning]  --> /checkout/tests/ui/imports/issue-28134.rs:3:10
   |
---
To only update this specific test, also pass `--test-args where-clauses/cfg_attribute.rs`

error in revision `a`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/where-clauses/cfg_attribute.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--cfg" "a" "--check-cfg" "cfg(test,FALSE,a,b)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/where-clauses/cfg_attribute.a" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: expected non-macro attribute, found attribute macro `derive`
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:32:7
   |
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:35:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:69:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:103:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:117:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:132:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:147:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:158:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:181:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:46:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:57:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:79:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:90:9
   |
LL |         #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:168:9
   |
LL |         #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: aborting due to 39 previous errors; 1 warning emitted
------------------------------------------

---- [ui] tests/ui/where-clauses/cfg_attribute.rs#a stdout end ----
---- [ui] tests/ui/where-clauses/cfg_attribute.rs#b stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/where-clauses/cfg_attribute.b/cfg_attribute.b.stderr`
diff of stderr:

76 LL |     #[derive(Clone)] ():,
77    |       ^^^^^^ not a non-macro attribute
78 
---
To only update this specific test, also pass `--test-args where-clauses/cfg_attribute.rs`

error in revision `b`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/where-clauses/cfg_attribute.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--cfg" "b" "--check-cfg" "cfg(test,FALSE,a,b)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/where-clauses/cfg_attribute.b" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: expected non-macro attribute, found attribute macro `derive`
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:32:7
   |
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:35:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:69:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:103:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:117:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:132:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:147:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:158:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:181:5
   |
LL |     #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:46:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:57:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:79:9
   |
LL |         #[rustfmt::skip] ():; //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:90:9
   |
LL |         #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/cfg_attribute.rs:168:9
   |
LL |         #[rustfmt::skip] ():, //~ ERROR most attributes are not supported in `where` clauses
   |         ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: aborting due to 39 previous errors; 1 warning emitted
------------------------------------------

---- [ui] tests/ui/where-clauses/cfg_attribute.rs#b stdout end ----
---- [ui] tests/ui/where-clauses/unsupported_attribute.rs stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/where-clauses/unsupported_attribute/unsupported_attribute.stderr`
diff of stderr:

10 LL |     #[derive(Clone)] 'a: 'static,
11    |       ^^^^^^ not a non-macro attribute
12 
+ warning: lint `soft_unstable` has been removed: the general soft-unstable mechanism has been removed
+   --> $DIR/unsupported_attribute.rs:8:10
+    |
---
To only update this specific test, also pass `--test-args where-clauses/unsupported_attribute.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/where-clauses/unsupported_attribute.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/where-clauses/unsupported_attribute" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: expected non-macro attribute, found attribute macro `derive`
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:28:7
   |
LL |     #[derive(Clone)] T: Trait,
   |       ^^^^^^ not a non-macro attribute

error: expected non-macro attribute, found attribute macro `derive`
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:31:7
   |
LL |     #[derive(Clone)] 'a: 'static,
   |       ^^^^^^ not a non-macro attribute

warning: lint `soft_unstable` has been removed: the general soft-unstable mechanism has been removed
##[warning]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:8:10
   |
---

error: `#[ignore]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:16:5
   |
LL |     #[ignore] T: Trait, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^
   |
   = help: `#[ignore]` can only be applied to functions

error: `#[ignore]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:17:5
   |
LL |     #[ignore] 'a: 'static, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^
   |
   = help: `#[ignore]` can only be applied to functions

error: `#[should_panic]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:18:5
   |
LL |     #[should_panic] T: Trait, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^^^
   |
   = help: `#[should_panic]` can only be applied to functions

error: `#[should_panic]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:19:5
   |
LL |     #[should_panic] 'a: 'static, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^^^
   |
   = help: `#[should_panic]` can only be applied to functions

error: `#[macro_use]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:20:5
   |
LL |     #[macro_use] T: Trait, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^
   |
   = help: `#[macro_use]` can be applied to crates, extern crates, and modules

error: `#[macro_use]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:21:5
   |
LL |     #[macro_use] 'a: 'static, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^
   |
   = help: `#[macro_use]` can be applied to crates, extern crates, and modules

error: `#[deprecated]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:24:5
   |
LL |     #[deprecated] T: Trait, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^
   |
   = help: `#[deprecated]` can be applied to associated consts, associated types, constants, crates, data types, enum variants, foreign statics, functions, inherent impl blocks, macro defs, modules, statics, struct fields, traits, type aliases, and use statements

error: `#[deprecated]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:25:5
   |
LL |     #[deprecated] 'a: 'static, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^
   |
   = help: `#[deprecated]` can be applied to associated consts, associated types, constants, crates, data types, enum variants, foreign statics, functions, inherent impl blocks, macro defs, modules, statics, struct fields, traits, type aliases, and use statements

error: `#[automatically_derived]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:26:5
   |
LL |     #[automatically_derived] T: Trait, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: `#[automatically_derived]` can only be applied to trait impl blocks

error: `#[automatically_derived]` attribute cannot be used on where predicates
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:27:5
   |
LL |     #[automatically_derived] 'a: 'static, //~ ERROR attribute cannot be used on
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: `#[automatically_derived]` can only be applied to trait impl blocks

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:14:5
   |
LL |     #[doc = "doc"] T: Trait, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:15:5
   |
LL |     #[doc = "doc"] 'a: 'static, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:22:5
   |
LL |     #[allow(unused)] T: Trait, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:23:5
   |
LL |     #[allow(unused)] 'a: 'static, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
---

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:31:5
   |
LL |     #[derive(Clone)] 'a: 'static,
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:34:5
   |
LL |     #[rustfmt::skip] T: Trait, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: most attributes are not supported in `where` clauses
##[error]  --> /checkout/tests/ui/where-clauses/unsupported_attribute.rs:35:5
   |
LL |     #[rustfmt::skip] 'a: 'static, //~ ERROR most attributes are not supported in `where` clauses
   |     ^^^^^^^^^^^^^^^^
   |
   = help: only `#[cfg]` and `#[cfg_attr]` are supported

error: aborting due to 20 previous errors; 1 warning emitted
```

</details>

--- a/compiler/rustc_attr_parsing/src/attributes/stability.rs
+++ b/compiler/rustc_attr_parsing/src/attributes/stability.rs
@@ -376,7 +376,6 @@ pub(crate) fn parse_unstability<S: Stage>(
     let mut reason = None;
     let mut issue = None;
     let mut issue_num = None;
-    let mut is_soft = false;
     let mut implied_by = None;
     let mut old_name = None;
 
@@ -423,12 +422,6 @@ pub(crate) fn parse_unstability<S: Stage>(
                     },
                 };
             }
-            Some(sym::soft) => {
-                if let Err(span) = args.no_args() {
-                    cx.emit_err(session_diagnostics::SoftNoArgs { span });
-                }
-                is_soft = true;
-            }
             Some(sym::implied_by) => {
                 insert_value_into_option_or_error(cx, &param, &mut implied_by, word.unwrap())?
             }
@@ -438,14 +431,7 @@ pub(crate) fn parse_unstability<S: Stage>(
             _ => {
                 cx.expected_specific_argument(
                     param.span(),
-                    &[
-                        sym::feature,
-                        sym::reason,
-                        sym::issue,
-                        sym::soft,
-                        sym::implied_by,
-                        sym::old_name,
-                    ],
+                    &[sym::feature, sym::reason, sym::issue, sym::implied_by, sym::old_name],
                 );
                 return None;
             }
@@ -468,7 +454,6 @@ pub(crate) fn parse_unstability<S: Stage>(
             let level = StabilityLevel::Unstable {
                 reason: UnstableReason::from_opt_reason(reason),
                 issue: issue_num,
-                is_soft,
                 implied_by,
                 old_name,
             };

--- a/compiler/rustc_attr_parsing/src/session_diagnostics.rs
+++ b/compiler/rustc_attr_parsing/src/session_diagnostics.rs
@@ -375,13 +375,6 @@ pub(crate) struct InvalidSince {
 }
 
 #[derive(Diagnostic)]
-#[diag("`soft` should not have any arguments")]
-pub(crate) struct SoftNoArgs {
-    #[primary_span]
-    pub span: Span,
-}
-
-#[derive(Diagnostic)]
 #[diag("unknown version literal format, assuming it refers to a future version")]
 pub(crate) struct UnknownVersionLiteral {
     #[primary_span]

--- a/compiler/rustc_hir/src/stability.rs
+++ b/compiler/rustc_hir/src/stability.rs
@@ -112,7 +112,6 @@ pub enum StabilityLevel {
         reason: UnstableReason,
         /// Relevant `rust-lang/rust` issue.
         issue: Option<NonZero<u32>>,
-        is_soft: bool,
         /// If part of a feature is stabilized and a new feature is added for the remaining parts,
         /// then the `implied_by` attribute is used to indicate which now-stable feature previously
         /// contained an item.

--- a/compiler/rustc_lint/src/lib.rs
+++ b/compiler/rustc_lint/src/lib.rs
@@ -642,6 +642,7 @@ fn register_builtins(store: &mut LintStore) {
          see <https://github.com/rust-lang/rust/issues/40107> for more information",
     );
     store.register_removed("wasm_c_abi", "the wasm C ABI has been fixed");
+    store.register_removed("soft_unstable", "the general soft-unstable mechanism has been removed");
 }
 
 fn register_internals(store: &mut LintStore) {

--- a/compiler/rustc_lint_defs/src/builtin.rs
+++ b/compiler/rustc_lint_defs/src/builtin.rs
@@ -105,7 +105,6 @@ declare_lint_pass! {
         SEMICOLON_IN_EXPRESSIONS_FROM_MACROS,
         SHADOWING_SUPERTRAIT_ITEMS,
         SINGLE_USE_LIFETIMES,
-        SOFT_UNSTABLE,
         STABLE_FEATURES,
         TAIL_EXPR_DROP_ORDER,
         TEST_UNSTABLE_LINT,
@@ -2314,22 +2313,6 @@ declare_lint! {
     "ambiguous associated items",
     @future_incompatible = FutureIncompatibleInfo {
         reason: fcw!(FutureReleaseError #57644),
-    };
-}
-
-declare_lint! {
-    /// The `soft_unstable` lint detects unstable features that were unintentionally allowed on
-    /// stable. This is a [future-incompatible] lint to transition this to a hard error in the
-    /// future. See [issue #64266] for more details.
-    ///
-    /// [issue #64266]: https://github.com/rust-lang/rust/issues/64266
-    /// [future-incompatible]: ../index.md#future-incompatible-lints
-    pub SOFT_UNSTABLE,
-    Deny,
-    "a feature gate that doesn't break dependent crates",
-    @future_incompatible = FutureIncompatibleInfo {
-        reason: fcw!(FutureReleaseError #64266),
-        report_in_deps: true,
     };
 }
 

--- a/compiler/rustc_middle/src/middle/stability.rs
+++ b/compiler/rustc_middle/src/middle/stability.rs
@@ -11,7 +11,7 @@ use rustc_hir::def_id::{DefId, LocalDefId};
 use rustc_hir::{self as hir, ConstStability, DefaultBodyStability, HirId, Stability};
 use rustc_macros::{Decodable, Encodable, HashStable, Subdiagnostic};
 use rustc_session::Session;
-use rustc_session::lint::builtin::{DEPRECATED, DEPRECATED_IN_FUTURE, SOFT_UNSTABLE};
+use rustc_session::lint::builtin::{DEPRECATED, DEPRECATED_IN_FUTURE};
 use rustc_session::lint::{BuiltinLintDiag, DeprecatedSinceKind, Level, Lint};
 use rustc_session::parse::feature_err_issue;
 use rustc_span::{Span, Symbol, sym};
@@ -68,9 +68,7 @@ pub fn report_unstable(
     reason: Option<Symbol>,
     issue: Option<NonZero<u32>>,
     suggestion: Option<(Span, String, String, Applicability)>,
-    is_soft: bool,
     span: Span,
-    soft_handler: impl FnOnce(&'static Lint, Span, String),
     kind: UnstableKind,
 ) {
     let qual = match kind {
@@ -83,18 +81,14 @@ pub fn report_unstable(
         None => format!("use of unstable{qual} library feature `{feature}`"),
     };
 
-    if is_soft {
-        soft_handler(SOFT_UNSTABLE, span, msg)
-    } else {
-        let mut err = feature_err_issue(sess, feature, span, GateIssue::Library(issue), msg);
-        if let Some((inner_types, msg, sugg, applicability)) = suggestion {
-            err.span_suggestion(inner_types, msg, sugg, applicability);
-        }
-        if let UnstableKind::Const(kw) = kind {
-            err.span_label(kw, "trait is not stable as const yet");
-        }
-        err.emit();
+    let mut err = feature_err_issue(sess, feature, span, GateIssue::Library(issue), msg);
+    if let Some((inner_types, msg, sugg, applicability)) = suggestion {
+        err.span_suggestion(inner_types, msg, sugg, applicability);
     }
+    if let UnstableKind::Const(kw) = kind {
+        err.span_label(kw, "trait is not stable as const yet");
+    }
+    err.emit();
 }
 
 fn deprecation_lint(is_in_effect: bool) -> &'static Lint {
@@ -266,7 +260,6 @@ pub enum EvalResult {
         reason: Option<Symbol>,
         issue: Option<NonZero<u32>>,
         suggestion: Option<(Span, String, String, Applicability)>,
-        is_soft: bool,
     },
     /// The item does not have the `#[stable]` or `#[unstable]` marker assigned.
     Unmarked,
@@ -386,7 +379,7 @@ impl<'tcx> TyCtxt<'tcx> {
 
         match stability {
             Some(Stability {
-                level: hir::StabilityLevel::Unstable { reason, issue, is_soft, implied_by, .. },
+                level: hir::StabilityLevel::Unstable { reason, issue, implied_by, .. },
                 feature,
                 ..
             }) => {
@@ -428,13 +421,7 @@ impl<'tcx> TyCtxt<'tcx> {
                 }
 
                 let suggestion = suggestion_for_allocator_api(self, def_id, span, feature);
-                EvalResult::Deny {
-                    feature,
-                    reason: reason.to_opt_reason(),
-                    issue,
-                    suggestion,
-                    is_soft,
-                }
+                EvalResult::Deny { feature, reason: reason.to_opt_reason(), issue, suggestion }
             }
             Some(_) => {
                 // Stable APIs are always ok to call and deprecated APIs are
@@ -469,7 +456,7 @@ impl<'tcx> TyCtxt<'tcx> {
 
         match stability {
             Some(DefaultBodyStability {
-                level: hir::StabilityLevel::Unstable { reason, issue, is_soft, .. },
+                level: hir::StabilityLevel::Unstable { reason, issue, .. },
                 feature,
             }) => {
                 if span.allows_unstable(feature) {
@@ -485,7 +472,6 @@ impl<'tcx> TyCtxt<'tcx> {
                     reason: reason.to_opt_reason(),
                     issue,
                     suggestion: None,
-                    is_soft,
                 }
             }
             Some(_) => {
@@ -563,30 +549,18 @@ impl<'tcx> TyCtxt<'tcx> {
         allow_unstable: AllowUnstable,
         unmarked: impl FnOnce(Span, DefId),
     ) -> bool {
-        let soft_handler = |lint, span, msg: String| {
-            self.emit_node_span_lint(
-                lint,
-                id.unwrap_or(hir::CRATE_HIR_ID),
-                span,
-                rustc_errors::DiagDecorator(|lint| {
-                    lint.primary_message(msg);
-                }),
-            );
-        };
         let eval_result =
             self.eval_stability_allow_unstable(def_id, id, span, method_span, allow_unstable);
         let is_allowed = matches!(eval_result, EvalResult::Allow);
         match eval_result {
             EvalResult::Allow => {}
-            EvalResult::Deny { feature, reason, issue, suggestion, is_soft } => report_unstable(
+            EvalResult::Deny { feature, reason, issue, suggestion } => report_unstable(
                 self.sess,
                 feature,
                 reason,
                 issue,
                 suggestion,
-                is_soft,
                 span,
-                soft_handler,
                 UnstableKind::Regular,
             ),
             EvalResult::Unmarked => unmarked(span, def_id),
@@ -623,12 +597,10 @@ impl<'tcx> TyCtxt<'tcx> {
 
         match stability {
             Some(ConstStability {
-                level: hir::StabilityLevel::Unstable { reason, issue, is_soft, implied_by, .. },
+                level: hir::StabilityLevel::Unstable { reason, issue, implied_by, .. },
                 feature,
                 ..
             }) => {
-                assert!(!is_soft);
-
                 if span.allows_unstable(feature) {
                     debug!("body stability: skipping span={:?} since it is internal", span);
                     return;
@@ -652,9 +624,7 @@ impl<'tcx> TyCtxt<'tcx> {
                     reason.to_opt_reason(),
                     issue,
                     None,
-                    false,
                     span,
-                    |_, _, _| {},
                     UnstableKind::Const(const_kw_span),
                 );
             }

--- a/compiler/rustc_passes/src/stability.rs
+++ b/compiler/rustc_passes/src/stability.rs
@@ -131,7 +131,6 @@ const FORCE_UNSTABLE: Stability = Stability {
     level: StabilityLevel::Unstable {
         reason: UnstableReason::Default,
         issue: NonZero::new(27812),
-        is_soft: false,
         implied_by: None,
         old_name: None,
     },

--- a/compiler/rustc_resolve/src/errors.rs
+++ b/compiler/rustc_resolve/src/errors.rs
@@ -1,7 +1,7 @@
 use rustc_errors::codes::*;
 use rustc_errors::formatting::DiagMessageAddArg;
 use rustc_errors::{
-    Applicability, Diag, DiagCtxtHandle, DiagMessage, Diagnostic, ElidedLifetimeInPathSubdiag,
+    Applicability, Diag, DiagCtxtHandle, Diagnostic, ElidedLifetimeInPathSubdiag,
     EmissionGuarantee, IntoDiagArg, Level, MultiSpan, Subdiagnostic, msg,
 };
 use rustc_macros::{Diagnostic, Subdiagnostic};
@@ -1452,17 +1452,6 @@ pub(crate) struct UnusedMacroDefinition {
 pub(crate) struct MacroRuleNeverUsed {
     pub n: usize,
     pub name: Symbol,
-}
-
-pub(crate) struct UnstableFeature {
-    pub msg: DiagMessage,
-}
-
-impl<'a> Diagnostic<'a, ()> for UnstableFeature {
-    fn into_diag(self, dcx: DiagCtxtHandle<'a>, level: Level) -> Diag<'a, ()> {
-        let Self { msg } = self;
-        Diag::new(dcx, level, msg)
-    }
 }
 
 #[derive(Diagnostic)]

--- a/compiler/rustc_resolve/src/macros.rs
+++ b/compiler/rustc_resolve/src/macros.rs
@@ -1064,8 +1064,7 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
     ) {
         let span = path.span;
         if let Some(stability) = &ext.stability
-            && let StabilityLevel::Unstable { reason, issue, is_soft, implied_by, .. } =
-                stability.level
+            && let StabilityLevel::Unstable { reason, issue, implied_by, .. } = stability.level
         {
             let feature = stability.feature;
 
@@ -1073,25 +1072,13 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                 |feature| self.tcx.features().enabled(feature) || span.allows_unstable(feature);
             let allowed_by_implication = implied_by.is_some_and(|feature| is_allowed(feature));
             if !is_allowed(feature) && !allowed_by_implication {
-                let lint_buffer = &mut self.lint_buffer;
-                let soft_handler = |lint, span, msg: String| {
-                    lint_buffer.buffer_lint(
-                        lint,
-                        node_id,
-                        span,
-                        // FIXME make this translatable
-                        errors::UnstableFeature { msg: msg.into() },
-                    )
-                };
                 stability::report_unstable(
                     self.tcx.sess,
                     feature,
                     reason.to_opt_reason(),
                     issue,
                     None,
-                    is_soft,
                     span,
-                    soft_handler,
                     stability::UnstableKind::Regular,
                 );
             }

--- a/compiler/rustc_span/src/symbol.rs
+++ b/compiler/rustc_span/src/symbol.rs
@@ -1914,7 +1914,6 @@ symbols! {
         slice_len_fn,
         slice_patterns,
         slicing_syntax,
-        soft,
         sparc,
         sparc64,
         sparc_target_feature,

--- a/src/librustdoc/clean/types.rs
+++ b/src/librustdoc/clean/types.rs
@@ -2430,7 +2430,7 @@ mod size_asserts {
     static_assert_size!(GenericParamDef, 40);
     static_assert_size!(Generics, 16);
     static_assert_size!(Item, 8);
-    static_assert_size!(ItemInner, 144);
+    static_assert_size!(ItemInner, 136);
     static_assert_size!(ItemKind, 48);
     static_assert_size!(PathSegment, 32);
     static_assert_size!(Type, 32);

--- a/tests/ui/feature-gates/issue-43106-gating-of-test.rs
+++ b/tests/ui/feature-gates/issue-43106-gating-of-test.rs
@@ -1,6 +1,5 @@
 // The non-crate level cases are in issue-43106-gating-of-builtin-attrs.rs.
 
-#![allow(soft_unstable)]
 #![test                    = "4200"]
 //~^ ERROR `test` attribute cannot be used at crate level
 fn main() {}

--- a/tests/ui/feature-gates/issue-43106-gating-of-test.stderr
+++ b/tests/ui/feature-gates/issue-43106-gating-of-test.stderr
@@ -1,5 +1,5 @@
 error: `test` attribute cannot be used at crate level
-  --> $DIR/issue-43106-gating-of-test.rs:4:1
+  --> $DIR/issue-43106-gating-of-test.rs:3:1
    |
 LL | #![test                    = "4200"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

--- a/tests/ui/imports/issue-28134.rs
+++ b/tests/ui/imports/issue-28134.rs
@@ -1,5 +1,4 @@
 //@ compile-flags: --test
 
-#![allow(soft_unstable)]
 #![test]
 //~^ ERROR `test` attribute cannot be used at crate level

--- a/tests/ui/imports/issue-28134.stderr
+++ b/tests/ui/imports/issue-28134.stderr
@@ -1,5 +1,5 @@
 error: `test` attribute cannot be used at crate level
-  --> $DIR/issue-28134.rs:4:1
+  --> $DIR/issue-28134.rs:3:1
    |
 LL | #![test]
    | ^^^^^^^^

--- a/tests/ui/where-clauses/cfg_attribute.a.stderr
+++ b/tests/ui/where-clauses/cfg_attribute.a.stderr
@@ -1,83 +1,83 @@
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:32:7
+  --> $DIR/cfg_attribute.rs:31:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:43:11
+  --> $DIR/cfg_attribute.rs:42:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:54:11
+  --> $DIR/cfg_attribute.rs:53:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:66:7
+  --> $DIR/cfg_attribute.rs:65:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:76:11
+  --> $DIR/cfg_attribute.rs:75:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:87:11
+  --> $DIR/cfg_attribute.rs:86:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:100:7
+  --> $DIR/cfg_attribute.rs:99:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:114:7
+  --> $DIR/cfg_attribute.rs:113:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:129:7
+  --> $DIR/cfg_attribute.rs:128:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:144:7
+  --> $DIR/cfg_attribute.rs:143:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:155:7
+  --> $DIR/cfg_attribute.rs:154:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:165:11
+  --> $DIR/cfg_attribute.rs:164:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:178:7
+  --> $DIR/cfg_attribute.rs:177:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:32:5
+  --> $DIR/cfg_attribute.rs:31:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -85,7 +85,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:35:5
+  --> $DIR/cfg_attribute.rs:34:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -93,7 +93,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:66:5
+  --> $DIR/cfg_attribute.rs:65:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -101,7 +101,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:69:5
+  --> $DIR/cfg_attribute.rs:68:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -109,7 +109,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:100:5
+  --> $DIR/cfg_attribute.rs:99:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -117,7 +117,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:103:5
+  --> $DIR/cfg_attribute.rs:102:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -125,7 +125,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:114:5
+  --> $DIR/cfg_attribute.rs:113:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -133,7 +133,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:117:5
+  --> $DIR/cfg_attribute.rs:116:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -141,7 +141,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:129:5
+  --> $DIR/cfg_attribute.rs:128:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -149,7 +149,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:132:5
+  --> $DIR/cfg_attribute.rs:131:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -157,7 +157,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:144:5
+  --> $DIR/cfg_attribute.rs:143:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -165,7 +165,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:147:5
+  --> $DIR/cfg_attribute.rs:146:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -173,7 +173,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:155:5
+  --> $DIR/cfg_attribute.rs:154:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -181,7 +181,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:158:5
+  --> $DIR/cfg_attribute.rs:157:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -189,7 +189,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:178:5
+  --> $DIR/cfg_attribute.rs:177:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -197,7 +197,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:181:5
+  --> $DIR/cfg_attribute.rs:180:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -205,7 +205,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:43:9
+  --> $DIR/cfg_attribute.rs:42:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -213,7 +213,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:46:9
+  --> $DIR/cfg_attribute.rs:45:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -221,7 +221,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:54:9
+  --> $DIR/cfg_attribute.rs:53:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -229,7 +229,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:57:9
+  --> $DIR/cfg_attribute.rs:56:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -237,7 +237,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:76:9
+  --> $DIR/cfg_attribute.rs:75:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -245,7 +245,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:79:9
+  --> $DIR/cfg_attribute.rs:78:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -253,7 +253,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:87:9
+  --> $DIR/cfg_attribute.rs:86:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -261,7 +261,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:90:9
+  --> $DIR/cfg_attribute.rs:89:9
    |
 LL |         #[rustfmt::skip] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -269,7 +269,7 @@ LL |         #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:165:9
+  --> $DIR/cfg_attribute.rs:164:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -277,7 +277,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:168:9
+  --> $DIR/cfg_attribute.rs:167:9
    |
 LL |         #[rustfmt::skip] ():,
    |         ^^^^^^^^^^^^^^^^

--- a/tests/ui/where-clauses/cfg_attribute.b.stderr
+++ b/tests/ui/where-clauses/cfg_attribute.b.stderr
@@ -1,83 +1,83 @@
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:32:7
+  --> $DIR/cfg_attribute.rs:31:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:43:11
+  --> $DIR/cfg_attribute.rs:42:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:54:11
+  --> $DIR/cfg_attribute.rs:53:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:66:7
+  --> $DIR/cfg_attribute.rs:65:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:76:11
+  --> $DIR/cfg_attribute.rs:75:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:87:11
+  --> $DIR/cfg_attribute.rs:86:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:100:7
+  --> $DIR/cfg_attribute.rs:99:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:114:7
+  --> $DIR/cfg_attribute.rs:113:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:129:7
+  --> $DIR/cfg_attribute.rs:128:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:144:7
+  --> $DIR/cfg_attribute.rs:143:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:155:7
+  --> $DIR/cfg_attribute.rs:154:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:165:11
+  --> $DIR/cfg_attribute.rs:164:11
    |
 LL |         #[derive(Clone)] ():,
    |           ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/cfg_attribute.rs:178:7
+  --> $DIR/cfg_attribute.rs:177:7
    |
 LL |     #[derive(Clone)] ():,
    |       ^^^^^^ not a non-macro attribute
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:32:5
+  --> $DIR/cfg_attribute.rs:31:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -85,7 +85,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:35:5
+  --> $DIR/cfg_attribute.rs:34:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -93,7 +93,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:66:5
+  --> $DIR/cfg_attribute.rs:65:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -101,7 +101,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:69:5
+  --> $DIR/cfg_attribute.rs:68:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -109,7 +109,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:100:5
+  --> $DIR/cfg_attribute.rs:99:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -117,7 +117,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:103:5
+  --> $DIR/cfg_attribute.rs:102:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -125,7 +125,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:114:5
+  --> $DIR/cfg_attribute.rs:113:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -133,7 +133,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:117:5
+  --> $DIR/cfg_attribute.rs:116:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -141,7 +141,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:129:5
+  --> $DIR/cfg_attribute.rs:128:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -149,7 +149,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:132:5
+  --> $DIR/cfg_attribute.rs:131:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -157,7 +157,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:144:5
+  --> $DIR/cfg_attribute.rs:143:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -165,7 +165,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:147:5
+  --> $DIR/cfg_attribute.rs:146:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -173,7 +173,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:155:5
+  --> $DIR/cfg_attribute.rs:154:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -181,7 +181,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:158:5
+  --> $DIR/cfg_attribute.rs:157:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -189,7 +189,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:178:5
+  --> $DIR/cfg_attribute.rs:177:5
    |
 LL |     #[derive(Clone)] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -197,7 +197,7 @@ LL |     #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:181:5
+  --> $DIR/cfg_attribute.rs:180:5
    |
 LL |     #[rustfmt::skip] ():,
    |     ^^^^^^^^^^^^^^^^
@@ -205,7 +205,7 @@ LL |     #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:43:9
+  --> $DIR/cfg_attribute.rs:42:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -213,7 +213,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:46:9
+  --> $DIR/cfg_attribute.rs:45:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -221,7 +221,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:54:9
+  --> $DIR/cfg_attribute.rs:53:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -229,7 +229,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:57:9
+  --> $DIR/cfg_attribute.rs:56:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -237,7 +237,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:76:9
+  --> $DIR/cfg_attribute.rs:75:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -245,7 +245,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:79:9
+  --> $DIR/cfg_attribute.rs:78:9
    |
 LL |         #[rustfmt::skip] ():;
    |         ^^^^^^^^^^^^^^^^
@@ -253,7 +253,7 @@ LL |         #[rustfmt::skip] ():;
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:87:9
+  --> $DIR/cfg_attribute.rs:86:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -261,7 +261,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:90:9
+  --> $DIR/cfg_attribute.rs:89:9
    |
 LL |         #[rustfmt::skip] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -269,7 +269,7 @@ LL |         #[rustfmt::skip] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:165:9
+  --> $DIR/cfg_attribute.rs:164:9
    |
 LL |         #[derive(Clone)] ():,
    |         ^^^^^^^^^^^^^^^^
@@ -277,7 +277,7 @@ LL |         #[derive(Clone)] ():,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/cfg_attribute.rs:168:9
+  --> $DIR/cfg_attribute.rs:167:9
    |
 LL |         #[rustfmt::skip] ():,
    |         ^^^^^^^^^^^^^^^^

--- a/tests/ui/where-clauses/cfg_attribute.rs
+++ b/tests/ui/where-clauses/cfg_attribute.rs
@@ -7,7 +7,6 @@
 #![feature(custom_test_frameworks)]
 #![feature(derive_const)]
 #![feature(where_clause_attrs)]
-#![allow(soft_unstable)]
 
 use std::marker::PhantomData;
 

--- a/tests/ui/where-clauses/unsupported_attribute.rs
+++ b/tests/ui/where-clauses/unsupported_attribute.rs
@@ -5,7 +5,6 @@
 #![feature(custom_test_frameworks)]
 #![feature(derive_const)]
 #![feature(where_clause_attrs)]
-#![allow(soft_unstable)]
 
 trait Trait {}
 

--- a/tests/ui/where-clauses/unsupported_attribute.stderr
+++ b/tests/ui/where-clauses/unsupported_attribute.stderr
@@ -1,17 +1,17 @@
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/unsupported_attribute.rs:28:7
+  --> $DIR/unsupported_attribute.rs:27:7
    |
 LL |     #[derive(Clone)] T: Trait,
    |       ^^^^^^ not a non-macro attribute
 
 error: expected non-macro attribute, found attribute macro `derive`
-  --> $DIR/unsupported_attribute.rs:31:7
+  --> $DIR/unsupported_attribute.rs:30:7
    |
 LL |     #[derive(Clone)] 'a: 'static,
    |       ^^^^^^ not a non-macro attribute
 
 error: `#[ignore]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:16:5
+  --> $DIR/unsupported_attribute.rs:15:5
    |
 LL |     #[ignore] T: Trait,
    |     ^^^^^^^^^
@@ -19,7 +19,7 @@ LL |     #[ignore] T: Trait,
    = help: `#[ignore]` can only be applied to functions
 
 error: `#[ignore]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:17:5
+  --> $DIR/unsupported_attribute.rs:16:5
    |
 LL |     #[ignore] 'a: 'static,
    |     ^^^^^^^^^
@@ -27,7 +27,7 @@ LL |     #[ignore] 'a: 'static,
    = help: `#[ignore]` can only be applied to functions
 
 error: `#[should_panic]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:18:5
+  --> $DIR/unsupported_attribute.rs:17:5
    |
 LL |     #[should_panic] T: Trait,
    |     ^^^^^^^^^^^^^^^
@@ -35,7 +35,7 @@ LL |     #[should_panic] T: Trait,
    = help: `#[should_panic]` can only be applied to functions
 
 error: `#[should_panic]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:19:5
+  --> $DIR/unsupported_attribute.rs:18:5
    |
 LL |     #[should_panic] 'a: 'static,
    |     ^^^^^^^^^^^^^^^
@@ -43,7 +43,7 @@ LL |     #[should_panic] 'a: 'static,
    = help: `#[should_panic]` can only be applied to functions
 
 error: `#[macro_use]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:20:5
+  --> $DIR/unsupported_attribute.rs:19:5
    |
 LL |     #[macro_use] T: Trait,
    |     ^^^^^^^^^^^^
@@ -51,7 +51,7 @@ LL |     #[macro_use] T: Trait,
    = help: `#[macro_use]` can be applied to crates, extern crates, and modules
 
 error: `#[macro_use]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:21:5
+  --> $DIR/unsupported_attribute.rs:20:5
    |
 LL |     #[macro_use] 'a: 'static,
    |     ^^^^^^^^^^^^
@@ -59,7 +59,7 @@ LL |     #[macro_use] 'a: 'static,
    = help: `#[macro_use]` can be applied to crates, extern crates, and modules
 
 error: `#[deprecated]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:24:5
+  --> $DIR/unsupported_attribute.rs:23:5
    |
 LL |     #[deprecated] T: Trait,
    |     ^^^^^^^^^^^^^
@@ -67,7 +67,7 @@ LL |     #[deprecated] T: Trait,
    = help: `#[deprecated]` can be applied to associated consts, associated types, constants, crates, data types, enum variants, foreign statics, functions, inherent impl blocks, macro defs, modules, statics, struct fields, traits, type aliases, and use statements
 
 error: `#[deprecated]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:25:5
+  --> $DIR/unsupported_attribute.rs:24:5
    |
 LL |     #[deprecated] 'a: 'static,
    |     ^^^^^^^^^^^^^
@@ -75,7 +75,7 @@ LL |     #[deprecated] 'a: 'static,
    = help: `#[deprecated]` can be applied to associated consts, associated types, constants, crates, data types, enum variants, foreign statics, functions, inherent impl blocks, macro defs, modules, statics, struct fields, traits, type aliases, and use statements
 
 error: `#[automatically_derived]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:26:5
+  --> $DIR/unsupported_attribute.rs:25:5
    |
 LL |     #[automatically_derived] T: Trait,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^
@@ -83,7 +83,7 @@ LL |     #[automatically_derived] T: Trait,
    = help: `#[automatically_derived]` can only be applied to trait impl blocks
 
 error: `#[automatically_derived]` attribute cannot be used on where predicates
-  --> $DIR/unsupported_attribute.rs:27:5
+  --> $DIR/unsupported_attribute.rs:26:5
    |
 LL |     #[automatically_derived] 'a: 'static,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^
@@ -91,7 +91,7 @@ LL |     #[automatically_derived] 'a: 'static,
    = help: `#[automatically_derived]` can only be applied to trait impl blocks
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:14:5
+  --> $DIR/unsupported_attribute.rs:13:5
    |
 LL |     #[doc = "doc"] T: Trait,
    |     ^^^^^^^^^^^^^^
@@ -99,7 +99,7 @@ LL |     #[doc = "doc"] T: Trait,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:15:5
+  --> $DIR/unsupported_attribute.rs:14:5
    |
 LL |     #[doc = "doc"] 'a: 'static,
    |     ^^^^^^^^^^^^^^
@@ -107,7 +107,7 @@ LL |     #[doc = "doc"] 'a: 'static,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:22:5
+  --> $DIR/unsupported_attribute.rs:21:5
    |
 LL |     #[allow(unused)] T: Trait,
    |     ^^^^^^^^^^^^^^^^
@@ -115,7 +115,7 @@ LL |     #[allow(unused)] T: Trait,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:23:5
+  --> $DIR/unsupported_attribute.rs:22:5
    |
 LL |     #[allow(unused)] 'a: 'static,
    |     ^^^^^^^^^^^^^^^^
@@ -123,7 +123,7 @@ LL |     #[allow(unused)] 'a: 'static,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:28:5
+  --> $DIR/unsupported_attribute.rs:27:5
    |
 LL |     #[derive(Clone)] T: Trait,
    |     ^^^^^^^^^^^^^^^^
@@ -131,7 +131,7 @@ LL |     #[derive(Clone)] T: Trait,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:31:5
+  --> $DIR/unsupported_attribute.rs:30:5
    |
 LL |     #[derive(Clone)] 'a: 'static,
    |     ^^^^^^^^^^^^^^^^
@@ -139,7 +139,7 @@ LL |     #[derive(Clone)] 'a: 'static,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:34:5
+  --> $DIR/unsupported_attribute.rs:33:5
    |
 LL |     #[rustfmt::skip] T: Trait,
    |     ^^^^^^^^^^^^^^^^
@@ -147,7 +147,7 @@ LL |     #[rustfmt::skip] T: Trait,
    = help: only `#[cfg]` and `#[cfg_attr]` are supported
 
 error: most attributes are not supported in `where` clauses
-  --> $DIR/unsupported_attribute.rs:35:5
+  --> $DIR/unsupported_attribute.rs:34:5
    |
 LL |     #[rustfmt::skip] 'a: 'static,
    |     ^^^^^^^^^^^^^^^^

